stripe/src/main/java/com/stripe/android/paymentsheet/analytics/DefaultDeviceIdRepository.kt

```diff
@@ -0,0 +1,38 @@
+package com.stripe.android.paymentsheet.analytics
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.core.content.edit
+import kotlinx.coroutines.withContext
+import kotlin.coroutines.CoroutineContext
+
+internal class DefaultDeviceIdRepository(
+    private val context: Context,
+    private val workContext: CoroutineContext
+) : DeviceIdRepository {
+    private val prefs: SharedPreferences by lazy {
+        context.getSharedPreferences(PREF_FILE, Context.MODE_PRIVATE)
+    }
+
+    override suspend fun get(): DeviceId = withContext(workContext) {
```

Comment:
I don't think `suspend` and `synchronized` can be used together. I made the shared prefs write use `commit` instead of `apply` so that it is written synchronously.